### ISSUE TITLE
Ghosts

### ISSUE THREAD

**Issue by alan-boggess** [CIS-499-Group3/Team3-Project#4]
From proposed deliverables:

"When a game is started, we will have a pacman-like figure on a map that may contain ghosts and pellets for eating.

...

In order to do this, we will need to be able to generate ... sprites for ... ghosts."

**Comment by SpenceSellers** [comment]
If we don't have ghosts working for real before the deliverables are due, it'll be about two lines of code to get a ghost sprite sitting there on the screen. It won't DO anything, but the deliverables didn't say anything about that ;)

**Comment by DaveWeathers** [comment]
Ok, simple blinky is in the assets.  32x32 etc

**Comment by SpenceSellers** [comment]
![Blinky now shows on screen](https://cloud.githubusercontent.com/assets/5377932/10298880/4b0312ee-6ba9-11e5-963f-b5b9d5d139fd.png)!

Deliverables technically done, the best type of done.

**Comment by DaveWeathers** [comment]
Animated blinky, he dances around in place, but nothing else.

**Comment by DaveWeathers** [comment]
Ok, going to get started on expanding out sprite collection, including ghosts, tomorrow.  Does anyone prefer that we do custom sprites?  or as close to the original as possible?  I suppose now is the time to decide.  I will likely start the ghosts from scratch either way as the originals are difficult to recolor properly.

**Comment by huntrben** [comment]
Given the disclaimer that we are "legally distinct", custom would probably be best. However, it's due in a week, so I'll defer to whatever you think is convenient

**Comment by alan-boggess** [comment]
Regarding "legal distinctness":  the _patent_ on the original Pac-Man games expired at least a dozen years ago.  I imagine Namco does still protect their _copyright_ for the actual games and maps and such that they produced, but as long as we aren't pulling assets directly out of another game, we're fine on the copyright front.

The "legally distinct" bit is meant sort of tongue-in-cheek, and it's not in the formal language of our deliverables.  As long as we aren't copying code or pulling assets from another source, we're as legally distinct as we need to be.  We can remove that line from the splash screen if we want, I just thought it would give our project a little character.

**Comment by alan-boggess** [comment]
So we've got a simple ghost behavior, now.  It's still pretty random, but it behaves enough like an opposing character that we could just implement four of them in each map and call it a day if we had to.

Our new deliverables promise "certain patterns of behavior" for ghosts.  How many behaviors do we want, and how "smart" do we want our ghosts to be?  I've implemented a Dijskstra-like algorithm before that I could probably drop here with a minimum of effort, but it will basically "explore" the entire map every time it wants to change direction, so it might be costly to execute.

**Comment by huntrben** [comment]
I like the behavior. that ghost moves around a lot. Once we add more ghosts and make them actually kill Pacman, we'll know if it becomes too hard. Although, too hard isn't a big deal so long as everything works

**Comment by alan-boggess** [comment]
Second behavior implemented.  Technically this fulfills the requirement of "certain patterns of behavior."  (Since we didn't promise a specific number.)

Dave's working on "reskinning" ghosts, so the next hurdle is going to be implementing ghost/player collision.

**Comment by DaveWeathers** [comment]
Ok, finished all the ghost sprites, will upload in a couple minutes.  I *believe* all necessary animation cases will be covered.

**Comment by DaveWeathers** [comment]
OK, ALL ghost animation sprites are uploaded (to my knowledge).  Please, give them a glance and make sure I didn't miss one.  As far as I know we have all the sprites we need.  I haven't made a cherry or any other fruit (aside from dots) because we hadn't talked about them.  I can, just let me know if that's what we want.  I can replace the large dots with cherries or w/e.

**Comment by SpenceSellers** [comment]
I made Clyde aggressively follow the player. Performance seems to be fine. He's a bit too dangerous right now, I'll probably change his speed to be 2/3 of the other ghosts or something.

**Comment by alan-boggess** [comment]
Just setting his speed to 145 instead of 150 should do the trick - epsilon centering doesn't add _that_ much speed, just enough to notice it.

**Comment by DaveWeathers** [comment]
Clyde is definitely brutal now.  And sitting still at the start point is no longer an option.

**Comment by huntrben** [comment]
Yeah. I've been messing around with speed, and I'm not sure I'm good enough to make it to the second map anymore.

**Comment by alan-boggess** [comment]
Well, it was an arcade game - it was designed to take your money as efficiently as possible.
